homeassistant/components/sensor/bunq.py

```diff
@@ -0,0 +1,214 @@
+        _LOGGER.error(err)
+        raise PlatformNotReady
+
+
+class BunqAccountSensor(Entity):
+    """Representation of a Bunq balance sensor."""
+
+    def __init__(self, account_name, account_id, currency, balance):
+        """Initialize the sensor."""
+        self._account_name = account_name
+        self._account_id = account_id
+        self._currency = currency
+        # since the call in setup to get accounts
+        # includes balances, we can initizalize with the state
+        self._state = balance
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._account_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._account_name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._currency
+
+    @property
+    def icon(self):
+        """Return the entity icon."""
+        return ICON
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    def load_data(self, data):
+        """Load the sensor with relevant data."""
+        try:
+            # store the current balance
+            old_balance = self._state
+            # write the new balance to state
+            self._state = data.get(self._account_id, None)
+            # return False if nothing has changed, true otherwise
+            return False if self._state == old_balance else True
+        except KeyError:
+            self._state = None
+            _LOGGER.warning('Count not find key in API results.')
+            return False
+
+
+class BunqData:
+    """Get the latest data and updates the states."""
+
+    def __init__(self, hass, bunq_context, bunq_accounts):
+        """Initialize the data object."""
+        self._bunq_context = bunq_context
+        self._bunq_accounts = bunq_accounts
+        self.data = {}
+        self.hass = hass
+
+    async def update_devices(self):
+        """Update all devices/sensors."""
+        if self._bunq_accounts:
+            tasks = []
+            # Update all devices
+            for acc in self._bunq_accounts:
+                # if the old and new balances where the same
+                # there is no need to call the state update method
+                if acc.load_data(self.data):
+                    tasks.append(acc.async_update_ha_state())
+            if tasks:
+                await asyncio.wait(tasks, loop=self.hass.loop)
+
+    async def schedule_update(self, minute=1):
+        """Schedule an update after minute minutes."""
+        _LOGGER.debug("Scheduling next update in {} minutes.".format(minute))
+        nxt = dt_util.utcnow() + timedelta(minutes=minute)
+        async_track_point_in_utc_time(self.hass, self.async_update, nxt)
```

Comment:
This works. You can also use `async_track_time_interval` or `async_call_later` to achieve similar results.